frontend/src/lib/k8s/apiProxy.test.ts

```diff
@@ -0,0 +1,1113 @@
+  });
+
+  describe('streamResults', () => {
+    it('Successfully starts a stream and receives data', done => {
+      apiProxy.streamResults(streamResultsUrl, cb, errCb, {
+        watch: '1',
+      });
+
+      mockServer.on('connection', async (socket: any) => {
+        expect(cb).toHaveBeenNthCalledWith(1, []);
+
+        socket.send(JSON.stringify({ type: 'ADDED', object: mockConfigMap }));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(cb).toHaveBeenNthCalledWith(2, [{ actionType: 'ADDED', ...mockConfigMap }]);
+
+        done();
+      });
+    });
+  });
+
+  describe('streamResultsForCluster', () => {
+    it('Successfully handles ADDED, MODIFIED, and DELETED types', done => {
+      apiProxy.streamResultsForCluster(
+        streamResultsUrl,
+        { cb, errCb, cluster: clusterName },
+        { watch: '1' }
+      );
+
+      mockServer.on('connection', async (socket: any) => {
+        expect(cb).toHaveBeenNthCalledWith(1, []);
+
+        socket.send(JSON.stringify({ type: 'ADDED', object: mockConfigMap }));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(cb).toHaveBeenNthCalledWith(2, [{ actionType: 'ADDED', ...mockConfigMap }]);
+
+        socket.send(JSON.stringify({ type: 'MODIFIED', object: modifiedConfigMap }));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(cb).toHaveBeenNthCalledWith(3, [{ actionType: 'MODIFIED', ...modifiedConfigMap }]);
+
+        socket.send(JSON.stringify({ type: 'DELETED', object: mockConfigMap }));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(cb).toHaveBeenNthCalledWith(4, []);
+
+        done();
+      });
+    });
+
+    it('Successfully handles error messages', done => {
+      apiProxy.streamResultsForCluster(
+        streamResultsUrl,
+        { cb, errCb, cluster: clusterName },
+        { watch: '1' }
+      );
+
+      mockServer.on('connection', async (socket: any) => {
+        socket.send(JSON.stringify(errorMessage));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(console.error).toHaveBeenCalledWith(
+          'Error in update',
+          expect.objectContaining({
+            object: expect.objectContaining({
+              actionType: errorMessage.object.actionType,
+              message: errorMessage.object.message,
+              reason: errorMessage.object.reason,
+            }),
+            type: errorMessage.type,
+          })
+        );
+        done();
+      });
+    });
+
+    it('Successfully handles stream cancellation', done => {
+      apiProxy
+        .streamResultsForCluster(
+          streamResultsUrl,
+          { cb, errCb, cluster: clusterName },
+          { watch: '1' }
+        )
+        .then(cancel => {
+          mockServer.on('connection', async socket => {
+            socket.send(JSON.stringify({ type: 'ADDED', object: mockConfigMap }));
+
+            setTimeout(() => {
```

Comment:
I left the timeouts in to be safe since there were some tests where we would not get a callback if we didn't wait, but I see now that some tests still pass without them. working on this now

Reply:
oops turns out most of them did not need them, I removed all the unnecessary ones and added comments for the ones that were. in each case, we would either get an empty callback or none at all, so the sleep calls are there to ensure we get proper callbacks